File: src/server/src/main/java/com/project/application/services/TagService.java
package com.project.application.services;

import com.project.application.models.tag.TagViewResource;
import com.project.domain.post.Tag;
import com.project.infrastructure.data.TagRepository;
import jakarta.inject.Singleton;

import java.util.List;
import java.util.UUID;

@Singleton
public class TagService {
    private final TagRepository tagRepository;

    public TagService(TagRepository tagRepository) {
        this.tagRepository = tagRepository;
    }

    public List<TagViewResource> getTags() {
        return this.tagRepository.findAll()
                .stream().map(t -> new TagViewResource(t.getId(), t.getName(), t.getTranslatedName()))
                .toList();
    }

    public List<Tag> getTags(List<UUID> tagIds) {
        return this.tagRepository.findByIdIn(tagIds);
    }
}

File: src/server/src/main/kotlin/com/project/application/services/TagService.kt
package com.project.application.services

import com.project.application.models.tag.TagViewResource
import com.project.domain.post.Tag
import com.project.infrastructure.data.TagRepository
import io.micronaut.transaction.annotation.Transactional
import jakarta.inject.Singleton
import java.util.UUID

@Singleton
open class TagService(private val tagRepository: TagRepository) {
    @Transactional(readOnly = true)
    open fun getTags(): List<TagViewResource> {
         return tagRepository.findAll()
                .map { t: Tag -> TagViewResource(t.id, t.name, t.translatedName) }
    }

    @Transactional(readOnly = true)
    open fun getTags(tagIds: List<UUID>): List<Tag> {
        return tagRepository.findByIdIn(tagIds)
    }
}
